internal/app/machined/pkg/controllers/k8s/kubelet_service.go

```diff
@@ -201,6 +212,25 @@ func (ctrl *KubeletServiceController) Run(ctx context.Context, r controller.Runt
 	}
 }
 
+// getHostname extracts the hostname from the kubelet args if specified or falls back to the system hostname.
+func (ctrl *KubeletServiceController) getHostname(kubeletArgs []string) (string, error) {
+	var hostname string
+
+	for _, arg := range kubeletArgs {
+		if strings.HasPrefix(arg, "--hostname-override=") {
+			hostname = strings.TrimPrefix(arg, "--hostname-override=")
+
+			break
+		}
+	}
+
+	if hostname != "" {
+		return hostname, nil
+	}
+
+	return os.Hostname()
```

Comment:
is it safe the get the hostname for the os package whereas we get it from the hostname resource elsewhere?

Reply:
I talked to @smira just now and we will do it differently - we'll add a new field to the `kubeletspec`.